modules/cluster/main.tf

```diff
@@ -23,13 +23,70 @@ resource "azurerm_kubernetes_cluster" "aks" {
 
   role_based_access_control {
     enabled = true
+  }
+}
 
-    azure_active_directory {
-      managed = true
-    }
+resource "azurerm_user_assigned_identity" "mi" {
+  name                = var.msi_name
+  location            = var.location
+  resource_group_name = var.resource_group_name
+}
+
+provider "helm" {
```

Comment:
Providers should be defined once and in the root modules  else we run in to provider conflicts in sub modules